pkg/apis/options/header.go

```diff
@@ -21,10 +21,10 @@ type Header struct {
 // make up the header value
 type HeaderValue struct {
 	// Allow users to load the value from a secret source
-	*SecretSource `json:",omitempty"`
+	*SecretSource `json:"secretSource,omitempty"`
 
 	// Allow users to load the value from a session claim
-	*ClaimSource `json:",omitempty"`
+	*ClaimSource `json:"claimSource,omitempty"`
```

Comment:
We probably want to call this something different, such as, `fromValue` with the previous field being `fromSecret`

Adding json tags to these is odd though since they are inlined, do you recall why this was needed?

I know on another PR we are talking about moving the prefix, is this the PR where in fact we should make this a discriminated union?

Reply:
They are inlined but because they are pointers mapstructures cannot resolve them properly. I wasn't able to get it to work otherwise, leading to this structure change:

![image](https://github.com/user-attachments/assets/4947aa41-673e-4805-82f1-41598968ad2e)

You have to explicitly mention if you want to use a claimSource or secretSource for each HeaderValue you set.